app/scripts/components/common/map/style-generators/wmts-timeseries.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import useFitBbox from '../hooks/use-fit-bbox';
+import { BaseGeneratorParams } from '../types';
+import { RasterPaintLayer } from './raster-paint-layer';
+import {
+  useRequestStatus,
+  useWebMapService
+} from '$components/common/map/style-generators/hooks';
+import { ActionStatus } from '$utils/status';
+import { userTzDate2utcString } from '$utils/date';
+
+export interface MapLayerWMTSProps extends BaseGeneratorParams {
+  id: string;
+  date?: Date;
+  stacCol: string;
+  sourceParams?: Record<string, any>;
+  stacApiEndpoint?: string;
+  zoomExtent?: number[];
+  onStatusChange?: (result: { status: ActionStatus; id: string }) => void;
+}
+
+export function WMTSTimeseries(props: MapLayerWMTSProps) {
+  const { id, stacCol, date, stacApiEndpoint, onStatusChange } = props;
+
+  const stacApiEndpointToUse = stacApiEndpoint ?? process.env.API_STAC_ENDPOINT;
+  const {
+    url: wmtsUrl,
+    bounds,
+    tilematrixSet
+  } = useWebMapService({
+    id,
+    stacCol,
+    stacApiEndpointToUse,
+    onStatusChange
+  });
+  const { changeStatus } = useRequestStatus({
+    id,
+    onStatusChange,
+    requestsToTrack: []
+  });
+  const tileParams = {
+    format: 'image/png',
+    Service: 'WMTS',
+    Request: 'GetTile',
+    Version: '1.0.0',
+    layer: stacCol,
+    style: 'default',
+    tilematrixset: tilematrixSet,
+    ...(date && { TIME: userTzDate2utcString(date) })
+  };
+  const primaryUrl = (wmtsUrl?.[0] ?? '') as string;
+
+  useFitBbox(false, undefined, bounds);
+
+  return (
+    <RasterPaintLayer
+      {...props}
+      id={id}
+      tileApiEndpoint={primaryUrl}
+      tileParams={tileParams}
+      generatorPrefix='wmts'
+      onStatusChange={changeStatus}
+      metadataFormatter={(_, tileParamsAsString) => ({
+        wmtsTileUrl: `${wmtsUrl}?${tileParamsAsString}`
```

Comment:
wmtsUrl is string[], should it be primaryUrl? fyi, this is called to load the tileurl for GIS. 

<img width="383" alt="Screenshot 2025-06-09 at 12 36 13 PM" src="https://github.com/user-attachments/assets/71b72027-29c0-4ec7-8efd-cdc9c1dd9841" />